backend/auth/handlers_login_test.go

```diff
@@ -0,0 +1,182 @@
+			inputData:    `{"username":"testuser", "password":"password123"}`,
+			expectedCode: http.StatusInternalServerError,
+			expectedBody: fmt.Sprintf(`{"error":"%s"}`, errorRetrieveUserAccount),
+		},
+		{
+			name:         "IncorrectPassword",
+			dbAdapter:    &MockMongoClientSuccess{},
+			inputData:    `{"username":"testuser", "password":"a-password"}`,
+			expectedCode: http.StatusUnauthorized,
+			expectedBody: fmt.Sprintf(`{"error":"%s"}`, errorIncorrectCredentials),
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			dbadapter.WebuiDBClient = tc.dbAdapter
+			req, err := http.NewRequest(http.MethodPost, "/login", strings.NewReader(tc.inputData))
+			if err != nil {
+				t.Fatalf("failed to create request: %v", err)
+			}
+			w := httptest.NewRecorder()
+
+			router.ServeHTTP(w, req)
+
+			if tc.expectedCode != w.Code {
+				t.Errorf("Expected `%v`, got `%v`", tc.expectedCode, w.Code)
+			}
+			if w.Body.String() != tc.expectedBody {
+				t.Errorf("Expected `%v`, got `%v`", tc.expectedBody, w.Body.String())
+			}
+		})
+	}
+}
+
+func TestLogin_SuccessCases(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	router := gin.Default()
+	mockJWTSecret := []byte("mockSecret")
+	router.Use(AuthMiddleware(mockJWTSecret))
+	AddService(router, mockJWTSecret)
+
+	testCases := []struct {
+		name             string
+		dbAdapter        dbadapter.DBInterface
+		inputData        string
+		expectedCode     int
+		expectedUsername string
+		expectedRole     int
+	}{
+		{
+			name:             "Success_AdminUser",
+			dbAdapter:        &MockMongoClientSuccess{},
+			inputData:        `{"username":"janedoe", "password":"password123!"}`,
+			expectedCode:     http.StatusOK,
+			expectedUsername: "janedoe",
+			expectedRole:     AdminRole,
+		},
+		{
+			name:             "Success_RegularUser",
+			dbAdapter:        &MockMongoClientRegularUser{},
+			inputData:        `{"username":"johndoe", "password":"password-123"}`,
+			expectedCode:     http.StatusOK,
+			expectedUsername: "johndoe",
+			expectedRole:     UserRole,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			dbadapter.WebuiDBClient = tc.dbAdapter
+			req, err := http.NewRequest(http.MethodPost, "/login", strings.NewReader(tc.inputData))
+			if err != nil {
+				t.Fatalf("failed to create request: %v", err)
+			}
+			w := httptest.NewRecorder()
+
+			router.ServeHTTP(w, req)
+
+			if tc.expectedCode != w.Code {
+				t.Errorf("Expected `%v`, got `%v`", tc.expectedCode, w.Code)
+			}
+			var responde_data map[string]string
+			err = json.Unmarshal([]byte(w.Body.Bytes()), &responde_data)
+			if err != nil {
+				t.Errorf("Unable to unmarshal response`%v`", w.Body.String())
+			}
+
+			response_token, exists := responde_data["token"]
+			if !exists {
+				t.Errorf("Unable to unmarshal response`%v`", w.Body.String())
+			}
+
+			token, parseErr := jwt.Parse(response_token, func(token *jwt.Token) (interface{}, error) {
```

Comment:
```suggestion
			token, parseErr := jwt.Parse(responseToken, func(token *jwt.Token) (interface{}, error) {
```
Suggesting to use Camel case.